core/workspace_comment_svg.js

```diff
@@ -0,0 +1,591 @@
+   * @private
+   */
+  this.useDragSurface_ =
+      Blockly.utils.is3dSupported() && !!workspace.blockDragSurface_;
+
+  Blockly.WorkspaceCommentSvg.superClass_.constructor.call(this,
+      workspace, content, height, width, opt_id);
+
+  this.render();
+}; goog.inherits(Blockly.WorkspaceCommentSvg, Blockly.WorkspaceComment);
+
+/**
+ * The width and height to use to size a workspace comment when it is first
+ * added, before it has been edited by the user.
+ * @type {number}
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.DEFAULT_SIZE = 100;
+
+/**
+ * Dispose of this comment.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.dispose = function() {
+  if (!this.workspace) {
+    // The comment has already been deleted.
+    return;
+  }
+  // If this comment is being dragged, unlink the mouse events.
+  if (Blockly.selected == this) {
+    this.unselect();
+    this.workspace.cancelCurrentGesture();
+  }
+
+  if (Blockly.Events.isEnabled()) {
+    Blockly.Events.fire(new Blockly.Events.CommentDelete(this));
+  }
+
+  goog.dom.removeNode(this.svgGroup_);
+  // Sever JavaScript to DOM connections.
+  this.svgGroup_ = null;
+  this.svgRect_ = null;
+  // Dispose of any rendered components
+  this.disposeInternal_();
+
+  Blockly.Events.disable();
+  Blockly.WorkspaceCommentSvg.superClass_.dispose.call(this);
+  Blockly.Events.enable();
+};
+
+/**
+ * Create and initialize the SVG representation of a workspace comment.
+ * May be called more than once.
+ * @package
+ */
+Blockly.WorkspaceCommentSvg.prototype.initSvg = function() {
+  goog.asserts.assert(this.workspace.rendered, 'Workspace is headless.');
+  if (!this.workspace.options.readOnly && !this.eventsInit_) {
+    Blockly.bindEventWithChecks_(
+        this.svgRectTarget_, 'mousedown', this, this.pathMouseDown_);
+    Blockly.bindEventWithChecks_(
+        this.svgHandleTarget_, 'mousedown', this, this.pathMouseDown_);
+  }
+  this.eventsInit_ = true;
+
+  this.updateMovable();
+  if (!this.getSvgRoot().parentNode) {
+    this.workspace.getBubbleCanvas().appendChild(this.getSvgRoot());
+  }
+};
+
+/**
+ * Handle a mouse-down on an SVG comment.
+ * @param {!Event} e Mouse down event or touch start event.
+ * @private
+ */
+Blockly.WorkspaceCommentSvg.prototype.pathMouseDown_ = function(e) {
+  var gesture = this.workspace.getGesture(e);
+  if (gesture) {
+    gesture.handleBubbleStart(e, this);
+  }
+};
+
+/**
+ * Show the context menu for this workspace comment.
+ * @param {!Event} e Mouse event.
+ * @private
+ */
+Blockly.WorkspaceCommentSvg.prototype.showContextMenu_ = function(e) {
+  if (this.workspace.options.readOnly) {
+    return;
+  }
+  // Save the current workspace comment in a variable for use in closures.
+  var comment = this;
+  var menuOptions = [];
+
+  if (this.isDeletable() && this.isMovable()) {
+    menuOptions.push(Blockly.ContextMenu.commentDuplicateOption(comment));
+    menuOptions.push(Blockly.ContextMenu.commentDeleteOption(comment));
```

Comment:
Should these be separated out into different checks? E.g. can a comment be moveable but not deletable or vice-versa?

Reply:
If it's not deletable we don't let you duplicate it (because then can you delete that one?) or delete it.

If it's not movable I guess maybe it's okay to allow duplication, but then can you move that comment?  Similarly, probably okay to allow deletion.  

This matches the logic for blocks in both blockly and scratch blocks.  If we want to change it, let's make that a separate issue and handle it for all of the cases.